src/main/java/io/vertx/jdbcclient/impl/JDBCPoolImpl.java

```diff
@@ -98,7 +98,7 @@ public void close(Handler<AsyncResult<Void>> handler) {
   @Override
   public <R> Future<R> schedule(ContextInternal contextInternal, CommandBase<R> commandBase) {
     ContextInternal ctx = vertx.getOrCreateContext();
-    return getConnectionInternal(ctx).flatMap(conn -> ((SqlConnectionImpl<?>) conn).schedule(ctx, commandBase).flatMap(r -> conn.close().map(r)));
+    return getConnectionInternal(ctx).flatMap(conn -> ((SqlConnectionImpl<?>) conn).schedule(ctx, commandBase).onFailure(err -> conn.close()).flatMap(r -> conn.close().map(r)));
```

Comment:
good catch, there is actually a specific future operation that handle this case that is eventually, so I think we could replace the `flatMap` with `eventually` and then there is no need for `onFailure`

Reply:
Done. Looks much nicer now...